### PR TITLE
Adding notes on how to setup test infrastructure for custom allocatio…

### PR DESCRIPTION
**Pull request by timtay-microsoft**
…n policy and far away iot hub

### CODE REVIEW

File: e2e/test/prerequisites/readme.md [hunk 1]
```diff
@@ -31,3 +31,26 @@ docker run -d --restart unless-stopped --name azure-iot-tinyproxy -p 127.0.0.1:8
...
+https://docs.microsoft.com/en-us/azure/iot-dps/how-to-use-custom-allocation-policies
+
+For the tests currently in this repo, your azure fucntion, which picks what iot hub to provision to, should always return the iot hub with the 
+longest hostname. Not all custom allocation policies need to work this way, but this is an easy-to-test functionality.
+
+In this folder, there is a run.csx file that is the code the azure function should run in order to achieve this behavior.
+
+The actual webhook url can be found through the azure portal, under your function, where there is a "Get Function URL" button that gives you a url 
+such as: "https://someazurefunction.azurewebsites.net/api/SomeTriggerName?code=XXXXXXXX" and this is the value that should be configured to the 
+CUSTOM_ALLOCATION_POLICY_WEBHOOK environment name
```
Comment (CIPop) [inline e2e/test/prerequisites/readme.md]: Please update the cmd and sh templates as with these new variables.

Comment (timtay-microsoft) [reply e2e/test/prerequisites/readme.md]: done

File: e2e/test/prerequisites/run.csx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+public class ResponseObj
+{
+    public string iotHubHostName {get; set;}
+    public string IoTHub {get; set;}
+    public DeviceTwinObj initialTwin {get; set;}
+    public string[] linkedHubs {get; set;}
+    public string enrollment {get; set;}
+}
```
Comment (CIPop) [inline e2e/test/prerequisites/run.csx]: Nit: missing NL at EOF.

File: e2e/test/prerequisites/run.csx [hunk 3]
```diff
@@ -0,0 +1,81 @@
+#r "Newtonsoft.Json"
```
Comment (CIPop) [inline e2e/test/prerequisites/run.csx]: Missing copyright header.
Also, is #r correct here?

Comment (timtay-microsoft) [reply e2e/test/prerequisites/run.csx]: Yes, the #r is an Azure function thing

**Pull request merged**